CASSANDRA-18185 Accumulate all `docs` PR

neshkeev opened the pull request:
I searched the entire repository for the `*.md` files and didn't spot any more of such typos.
Here is the command:

```bash
find . -not -path "./.git/*" -type f -name "*.md" -exec grep --color "\(\b\w\+\)\b\s*\1\b" {} \+
```

neshkeev commented:
@smiklosovic , could you help me find someone who can merge the PR, please?

smiklosovic commented:
Hi @neshkeev 

thanks for the patch. I think it will be better if we put all doc-related changes together in one big patch and merge that one. It is not too much "effective" to merge this very tiny change. It would be way more preferable (from my point of view) to create a patch which would gather all docs fixes among pull requests and merge that one and we are ok for good.

I think there is a lot of docs PRs (if you filter them by "docs" tag), some of them might be obsolete and somebody would need to go through them.

neshkeev commented:
I'll peruse the PRs with such a tag and try to combine them into a single one if possible.
Should I somehow include original authors of the PRs I'll combine into a single PR?

smiklosovic commented:
Preferably.

You do that by specifying `Co-authored-by: ...` in the commit message.

smiklosovic commented:
@neshkeev if you go through other PRs related to docs, ping me if you think it is not relevant anymore I will double check and we can close it.

neshkeev committed:
[CASSANDRA-18185] Accumulate all the small typo fixes

This patch includes all the changes from the PRs that introduce small
changes related to typos in the documentation. The changes are
accumulated from the followig PRs:

- https://github.com/apache/cassandra/pull/206
- https://github.com/apache/cassandra/pull/359
- https://github.com/apache/cassandra/pull/366
- https://github.com/apache/cassandra/pull/390
- https://github.com/apache/cassandra/pull/450
- https://github.com/apache/cassandra/pull/567
- https://github.com/apache/cassandra/pull/615
- https://github.com/apache/cassandra/pull/618
- https://github.com/apache/cassandra/pull/746
- https://github.com/apache/cassandra/pull/974
- https://github.com/apache/cassandra/pull/984
- https://github.com/apache/cassandra/pull/1052
- https://github.com/apache/cassandra/pull/1088
- https://github.com/apache/cassandra/pull/1274
- https://github.com/apache/cassandra/pull/1378
- https://github.com/apache/cassandra/pull/1404
- https://github.com/apache/cassandra/pull/1504
- https://github.com/apache/cassandra/pull/1540
- https://github.com/apache/cassandra/pull/1544
- https://github.com/apache/cassandra/pull/1673
- https://github.com/apache/cassandra/pull/1697
- https://github.com/apache/cassandra/pull/1722
- https://github.com/apache/cassandra/pull/1815
- https://github.com/apache/cassandra/pull/1830
- https://github.com/apache/cassandra/pull/1863
- https://github.com/apache/cassandra/pull/1865
- https://github.com/apache/cassandra/pull/1879
- https://github.com/apache/cassandra/pull/2062

Co-authored-by: kalmant <kalmant@users.noreply.github.com>
Co-authored-by: Dmitry <xotonic@yandex.ru>
Co-authored-by: Tibor Répási <rtib@users.noreply.github.com>
Co-authored-by: Tzach Livyatan <tzach@scylladb.com>
Co-authored-by: Jérôme BAROTIN <jeromebarotin@gmail.com>
Co-authored-by: Giorgio Giuffrè <giorgiogiuffre23@gmail.com>
Co-authored-by: Siddhartha Tiwari <201851127@iiitvadodara.ac.in>
Co-authored-by: Angelo Polo <language.devel@gmail.com>
Co-authored-by: Tjeu Kayim <15987676+TjeuKayim@users.noreply.github.com>
Co-authored-by: Hemang Ajmera <81636438+hemangajmera@users.noreply.github.com>
Co-authored-by: 陳傑夫 <chienfuchen32@gmail.com>
Co-authored-by: Bhouse99 <bhouse99@protonmail.com>
Co-authored-by: Matthew Hardwick <MatthewRHardwick@gmail.com>
Co-authored-by: Paul Wouters <paul.wouters@aiven.io>
Co-authored-by: Romain Hardouin <romain_hardouin@yahoo.fr>
Co-authored-by: Romain Hardouin <romain_hardouin@yahoo.fr>
Co-authored-by: Guilherme Poleto <gpoleto@alunos.utfpr.edu.br>
Co-authored-by: 陳傑夫 <chienfuchen32@gmail.com>
Co-authored-by: etc-crontab <jujut@free.fr>
Co-authored-by: etc-crontab <jujut@free.fr>
Co-authored-by: etc-crontab <jujut@free.fr>
Co-authored-by: Prashant Bhuruk <prashantbhuruk88@gmail.com>
Co-authored-by: Jingchuan Zhu <56401528+codingswag998@users.noreply.github.com>
Co-authored-by: Ryan Stewart <ryan.stewart@rackspace.com>
Co-authored-by: utkarsh-agrawal-jm <107914361+utkarsh-agrawal-jm@users.noreply.github.com>
Co-authored-by: Ben Dalling <b.dalling@locp.co.uk>
Co-authored-by: Terry L. Blessing <tlblessing1@gmail.com>
Co-authored-by: gruzilkin <gruzmob@gmail.com>
Co-authored-by: gruzilkin <gruzmob@gmail.com>
Co-authored-by: Kevin <kevin.xgr@gmail.com>
Co-authored-by: yziadeh <121903189+yziadeh@users.noreply.github.com>
Signed-off-by: Nikita Eshkeev <neshkeev@yandex.ru>

neshkeev commented:
@smiklosovic , I reviewed all the `docs` labeled PRs starting from [#206](https://github.com/apache/cassandra/pull/206) and joined them into a single commit here.

The following PRs weren't merged, because they are no longer relevant, either files were removed (completely, I tracked files that were renamed like `*.rst` -> `*.adoc`) or the fixes they proposed were already applied in different commits.

PRs that require attention:

https://github.com/apache/cassandra/pull/1655 - `gen-nodetool-docs.py` script is proposed to get changed in that PR
https://github.com/apache/cassandra/pull/589 - `cqlsh.py` script is proposed to get changed in that PR

PRs that can be safely closed without merging, because I merged them into this PR's commit:

https://github.com/apache/cassandra/pull/206
https://github.com/apache/cassandra/pull/359
https://github.com/apache/cassandra/pull/366
https://github.com/apache/cassandra/pull/390
https://github.com/apache/cassandra/pull/450
https://github.com/apache/cassandra/pull/567
https://github.com/apache/cassandra/pull/615
https://github.com/apache/cassandra/pull/618
https://github.com/apache/cassandra/pull/746
https://github.com/apache/cassandra/pull/974
https://github.com/apache/cassandra/pull/984
https://github.com/apache/cassandra/pull/1052
https://github.com/apache/cassandra/pull/1088
https://github.com/apache/cassandra/pull/1274
https://github.com/apache/cassandra/pull/1378
https://github.com/apache/cassandra/pull/1404
https://github.com/apache/cassandra/pull/1504
https://github.com/apache/cassandra/pull/1540
https://github.com/apache/cassandra/pull/1544
https://github.com/apache/cassandra/pull/1673
https://github.com/apache/cassandra/pull/1697
https://github.com/apache/cassandra/pull/1722
https://github.com/apache/cassandra/pull/1815
https://github.com/apache/cassandra/pull/1830
https://github.com/apache/cassandra/pull/1863
https://github.com/apache/cassandra/pull/1865
https://github.com/apache/cassandra/pull/1879
https://github.com/apache/cassandra/pull/2062

PRs that wasn't merged into this PR's commit, but still can be closed without merging:

https://github.com/apache/cassandra/pull/456  - a complex PR about data modeling, I wasn't sure I could merge it appropriately
https://github.com/apache/cassandra/pull/648 - already fixed in a different commit
https://github.com/apache/cassandra/pull/703 - already fixed in a different commit
https://github.com/apache/cassandra/pull/842 - the file has been removed
https://github.com/apache/cassandra/pull/888 - the file has been removed
https://github.com/apache/cassandra/pull/944 - already fixed in a different commit
https://github.com/apache/cassandra/pull/1073 - already fixed in a different commit
https://github.com/apache/cassandra/pull/1088 - the file has been removed
https://github.com/apache/cassandra/pull/1600 - already fixed in a different commit
https://github.com/apache/cassandra/pull/1808 - already fixed in a different commit
https://github.com/apache/cassandra/pull/1809 - already fixed in a different commit
https://github.com/apache/cassandra/pull/1810 - already fixed in a different commit
https://github.com/apache/cassandra/pull/1829 - already fixed in a different commit
https://github.com/apache/cassandra/pull/1861 - already fixed in a different commit
https://github.com/apache/cassandra/pull/1862 - already fixed in a different commit
https://github.com/apache/cassandra/pull/1864 - already fixed in a different commit
https://github.com/apache/cassandra/pull/1880 - already fixed in a different commit
https://github.com/apache/cassandra/pull/1943 - already fixed in a different commit

michaelsembwever commented:
Generally LGTM.

But… 
A more descriptive commit message please. 
"Accumulate all docs PR" means nothing to me.

Something like "Accumulation of small typo fixes to docs" would be much better. Folk need to scan git history quickly.

I'm also not sure about the changes to hyphenation (coming from [this PR](https://github.com/apache/cassandra/pull/567/)).  For example "open-source" just reads wrong to me (even when it's used as an adjective.  The overall change to using hyphenation might need broader agreement. Personally I would just leave it out to move ahead with the other changes.

neshkeev commented:
Those hyphenated adjectives looked odd to me too, when I was in doubt, I googled them, for example, [open-source databases](https://phoenixnap.com/kb/open-source-database), or [5-node cluster](https://www.servethehome.com/building-5-node-proxmox-ve-cluster-20-minutes-dollar-phoenixnap-intel-supermicro/)

I agree with you, we need more input on this from the community to find out the best possible phrasing

smiklosovic commented:
great work @neshkeev , I will take a closer look shortly.

smiklosovic commented on conf/cqlshrc.sample:
```diff
@@ -109,6 +112,9 @@ port = 9042
 
 
 ;[ssl]
+;; Version of TLS to be used. Required to be set for cassandra version > 3.11.8
```
@neshkeev I do not think this is necessary to add, there is this later in `ssl` section:

````
; this is effectively ignored from 4.1 included as TLS protocol is auto-negotiated and will
; be removed in the next major version of Cassandra, possible values were TLSv1, TLSv1_1 or TLSv1_2
;version =
````

neshkeev replied:
I removed it

smiklosovic replied:
https://issues.apache.org/jira/browse/CASSANDRA-17452

smiklosovic commented on conf/cqlshrc.sample:
```diff
@@ -33,6 +32,10 @@
 ; classname = PlainTextAuthProvider
 ; username = user1
 
+[protocol]
```
@neshkeev is this really relevant? can you point me to the PR suggesting to add this?

smiklosovic commented on conf/cqlshrc.sample:
```diff
@@ -33,6 +32,10 @@
 ; username = user1
 
+[protocol]
+;; Specify a specific protcol version otherwise the client will default and downgrade as necessary
```
@neshkeev is this really relevant? can you point me to the PR suggesting to add this?

neshkeev replied:
@smiklosovic , [here](https://github.com/apache/cassandra/pull/366/files#diff-f55a268d347d301121adb03b1d33958837ed984295b83fbe474dde2781fe809fR26) it is

smiklosovic commented on doc/modules/cassandra/pages/architecture/guarantees.adoc:
```diff
@@ -1,17 +1,21 @@
 
 Apache Cassandra is a highly scalable and reliable database. Cassandra
-is used in web based applications that serve large number of clients and
+<<<<<<< HEAD
```
@neshkeev this seems like conflicts were not resolved properly.

neshkeev replied:
It's fixed

smiklosovic commented on doc/modules/cassandra/pages/architecture/guarantees.adoc:
```diff
@@ -47,46 +51,50 @@ Cassandra makes the following guarantees.
 multi-datacenter environment the replicas may be stored on different
 datacenters. If one replica is lost due to unrecoverable node/datacenter
-failure the data is not completely lost as replicas are still available.
+<<<<<<< HEAD
```
@neshkeev not resolved conflicts

neshkeev replied:
It's fixed

smiklosovic commented on doc/modules/cassandra/pages/faq/index.adoc:
```diff
@@ -79,7 +79,10 @@ intensive process that may result in adverse cluster performance. It's
 entire cluster at once will most likely swamp it. Note that you will
 need to run a full repair (`-full`) to make sure that already repaired
-sstables are not skipped.
+sstables are not skipped. You should use ConsistencyLevel.QUORUM or ALL
```
may be ... `ConsistencyLevel.QUORUM` or `ALL` ...

neshkeev replied:
It's fixed

smiklosovic commented on doc/modules/cassandra/pages/tools/cqlsh.adoc:
```diff
@@ -38,7 +38,7 @@ modules that are central to the performance of `COPY`.
 
 The `cqlshrc` file holds configuration options for `cqlsh`. 
-By default, the file is locagted the user's home directory at `~/.cassandra/cqlsh`, but a
+By default, the file is locagted the user's home directory at `~/.cassandra/cqlshrc`, but a
```
`locagted` -> `located`

neshkeev replied:
Good catch! I fixed it

smiklosovic commented on src/java/org/apache/cassandra/service/StartupChecks.java:
```diff
@@ -226,7 +226,7 @@ public void execute(StartupChecksOptions options)
     {
         /**
-         * The earliest legit timestamp a casandra instance could have ever launched.
+         * The earliest legit timestamp a cassandra instance could have ever launched.
```
`cassandra` -> `Cassandra`

neshkeev replied:
Fixed

smiklosovic commented:
@neshkeev I believe PR 1073 is still relevant if you look closer.

neshkeev commented:
> @neshkeev I believe PR 1073 is still relevant if you closer.

The changes it proposes were proposed in the [#1052](https://github.com/apache/cassandra/pull/1052/files#diff-a63e202ba3c4f0ac0e2667dc30d7bba293011ed9917be8e2244e8f40cf4bb445R386) PR and they are in [the current PR](https://github.com/apache/cassandra/pull/2062/files#diff-7b9caadf725c996c57188d3403035f79bdff83dc12dd510da5faf970d0ec9cd2R455)

smiklosovic commented:
@neshkeev could you please specifically check this one? https://github.com/apache/cassandra/pull/1052/files#r1083818544

smiklosovic commented:
I believe that #589 and #1655 can be worked on independently from this one. Especially if these patches are planning to be merged in various branches and this targets just trunk.

smiklosovic commented:
@neshkeev great progress so far! what about cracking thought the rest after (or before) PR 206 as you mentioned?

neshkeev commented:
> what about cracking thought the rest after (or before) PR 206 as you mentioned?

Sorry, I didn't get that, could you, please rephrase it?

All the changes https://github.com/apache/cassandra/pull/206 proposes are included into this PR, or were fixed before me in different commits, only the [`impact` -> `impacts`](https://github.com/apache/cassandra/pull/206/files#diff-6a259516d12c778b2b08ec408b884a88bf94d14a9f6dd8d943c6e4376288da6dR346) typo is left till now

smiklosovic commented:
@neshkeev You wrote:

_I reviewed all the docs labeled PRs starting from https://github.com/apache/cassandra/pull/206 and joined them into a single commit here._ 

So, it means that all PRs older than PR 206 (PRs 206 -> 1) are _not reviewed yet_ ? If that is the case, could we review them all?

neshkeev commented:
I should've emphasize it earlier: 206 is the first (chronologically) PR that is labeled with `docs`, there are no `docs` PRs before 206

smiklosovic commented:
right ... so I think this is the last one as I put that comment there:

https://github.com/apache/cassandra/pull/180#issuecomment-1400104542

Thanks!

smiklosovic commented:
@neshkeev could  you please cherry-pick this?

https://github.com/instaclustr/cassandra/commit/20194117b5371d8e9d6f0673bb0a685262dc739e

I think we are in quite a good shape, I ll try to find somebody else to take a look too.

Mmuzaf commented:
@neshkeev Please, consider the following typo fixes.
https://github.com/Mmuzaf/cassandra/commit/94452527b9a3f963871a5839af5e841b4e21af76

smiklosovic commented:
I am not completely sure about these typos in java classes, they might probably go in as well but we should ask. It is categorically something different to commit "documentation" and "fix typos in sources".

polandll commented on doc/modules/cassandra/pages/architecture/dynamo.adoc:
```diff
@@ -256,7 +255,7 @@ secondary indices with them.
 for production use. The expected audience is experienced users of
 Cassandra capable of fully validating a deployment of their particular
-application. That means being able check that operations like reads,
+application. That means being able to check that operations like reads,
```
```suggestion
application. That means you have the experience to check that operations like reads,
```

polandll commented on doc/modules/cassandra/pages/architecture/guarantees.adoc:
```diff
@@ -97,12 +97,12 @@ uncommitted, without making a new addition or update.
-event the coordinator node fails.
+stored in the cluster, the batchlog data is removed. The batch is
+replicated to another node to ensure that the full batch completes in
+the event if the coordinator node fails.
```
```suggestion
the event the coordinator node fails.
```

polandll commented on doc/modules/cassandra/pages/architecture/overview.adoc:
```diff
@@ -59,19 +59,19 @@ keys.
+* Collection types including sets, maps, and lists
 * Local secondary indices
-* (Experimental) materialized views
+* (Experimental) materialized views.
```
```suggestion
* (Experimental) materialized views
```

polandll commented on doc/modules/cassandra/pages/architecture/snitch.adoc:
```diff
@@ -2,17 +2,17 @@
 In cassandra, the snitch has two functions:
 
-* it teaches Cassandra enough about your network topology to route
+* It teaches Cassandra enough about your network topology to route
```
Generally, with a colon in the previous sentence, you use lowercase to start the bullet item.

polandll commented on doc/modules/cassandra/pages/architecture/snitch.adoc:
```diff
@@ -2,17 +2,17 @@
+* It teaches Cassandra enough about your network topology to route
 requests efficiently.
-* it allows Cassandra to spread replicas around your cluster to avoid
+* It allows Cassandra to spread replicas around your cluster to avoid
```
same comment

polandll commented on doc/modules/cassandra/pages/architecture/storage_engine.adoc:
```diff
@@ -3,17 +3,17 @@
+limited by the `commitlog_segment_size` option. Once the size is
 reached, a new commitlog segment is created. Commitlog segments can be
-archived, deleted, or recycled once all its data has been flushed to
+archived, deleted, or recycled once all their data has been flushed to
```
```suggestion
archived, deleted, or recycled once all the data has been flushed to
```

polandll replied:
Don't anthropomorphize. a commit log is not a person. :-)

polandll commented on doc/modules/cassandra/pages/architecture/storage_engine.adoc:
```diff
@@ -221,5 +224,5 @@ match the "ib" SSTable version
 [source,bash]
 ----
-include:example$find_sstables.sh[]
+include::../../examples/BASH/find_sstables.sh[]
```
```suggestion
include::example$BASH/find_sstables.sh[]
```

polandll replied:
Never use a path in includes in asciidoc/antora. https://docs.antora.org/antora/latest/page/include-an-example/

polandll commented on doc/modules/cassandra/pages/cql/types.adoc:
```diff
@@ -338,13 +338,13 @@ include::example$CQL/update_list.cql[]
 ====
 The append and prepend operations are not idempotent by nature. So in
-particular, if one of these operation timeout, then retrying the
+particular, if one of these operations timeout, then retrying the
```
```suggestion
particular, if one of these operations times out, then retrying the
```

polandll commented on doc/modules/cassandra/pages/operating/auditlogging.adoc:
```diff
@@ -88,10 +88,23 @@ Common audit log entry types are one of the following:
+For example, an ``INSERT`` in a keyspace with replication factor of 3 will produce an audit entry on one node, the coordinator who handled the request, and not on the two other nodes.
+For this reason, and depending on compliance requirements you must meet, you have to make sure that audig logs are stored on a non-ephemeral storage.
+
+You can achieve custom needs with <<archive_command>> option.
```
```suggestion
You can achieve custom needs with the <<archive_command>> option.
```

polandll commented on doc/modules/cassandra/pages/operating/auditlogging.adoc:
```diff
@@ -88,10 +88,23 @@ Common audit log entry types are one of the following:
+
+For a given query, the corresponding audit entry is only stored on the coordinator node.
+For example, an ``INSERT`` in a keyspace with replication factor of 3 will produce an audit entry on one node, the coordinator who handled the request, and not on the two other nodes.
+For this reason, and depending on compliance requirements you must meet, you have to make sure that audig logs are stored on a non-ephemeral storage.
```
```suggestion
For this reason, and depending on compliance requirements you must meet, make sure that audit logs are stored on a non-ephemeral storage.
```

polandll commented on doc/modules/cassandra/pages/operating/auditlogging.adoc:
```diff
@@ -123,16 +136,25 @@ audit_logging_options:
 
-Audit logging is enabled by setting the `enabled` option to `true` in
-the `audit_logging_options` setting. 
+Control wether audit logging is enabled or disabled (default).
```
```suggestion
Control whether audit logging is enabled or disabled (default).
```

polandll commented on doc/modules/cassandra/pages/operating/auditlogging.adoc:
```diff
@@ -186,52 +210,101 @@ excluded_categories: DDL, DML, QUERY, PREPARE
+
+For example: ``block: false`` to drop records (e.g. if audit is used for troobleshooting)
+
+For regulatory compliance purpose, it's a good practice to explicitely set ``block: true`` to prevent any regression in case of future default value change.
```
```suggestion
For regulatory compliance purposes, it's a good practice to explicitly set ``block: true`` to prevent any regression in case of future default value change.
```

polandll commented on doc/modules/cassandra/pages/operating/auditlogging.adoc:
```diff
@@ -186,52 +210,101 @@ excluded_categories: DDL, DML, QUERY, PREPARE
 
-The ``max_archive_retries`` option sets the max number of retries of failed archive commands. The default is 10.
-For example: ``max_archive_retries: 10``
+When using a user-defined script, Cassandra do **not** use the DeletingArchiver, so it's the responsability of the script to make any required cleanup.
```
```suggestion
When using a user-defined script, Cassandra does **not** use the DeletingArchiver, so it's the responsibility of the script to make any required cleanup.
```

polandll commented on doc/modules/cassandra/pages/operating/auditlogging.adoc:
```diff
@@ -186,52 +210,101 @@ excluded_categories: DDL, DML, QUERY, PREPARE
+----
 
-Audit logging can also be configured using ``nodetool` when enabling the feature, and will override any values set in the `cassandra.yaml` file, as discussed in the next section.
+TIP: Because Cassandra does not make use of Pretoucher, you can configure Chronicle Queue to shrink files synchronously -- i.e. as soon as the file is rolled -- with ``chronicle.queue.synchronousFileShrinking`` JVM properties. For instance, you can add the following line at the end of ``cassandra-env.sh``: ``JVM_OPTS="$JVM_OPTS -Dchronicle.queue.synchronousFileShrinking=true"``
```
```suggestion
TIP: 
====
Because Cassandra does not make use of Pretoucher, you can configure Chronicle Queue to shrink files synchronously -- i.e. as soon as the file is rolled -- with `chronicle.queue.synchronousFileShrinking` JVM properties. For instance, you can add the following line at the end of `cassandra-env.sh`: `JVM_OPTS="$JVM_OPTS -Dchronicle.queue.synchronousFileShrinking=true"`
====

smiklosovic replied:
This does not render it correctly in IDEA. I left it unchanged.

michaelsembwever replied:
Does it render correctly when generated? (that really is all we care about – i suspect thinking IDEA rendering helps traps us)

polandll reviewed (changes requested):
There are some changes that should be made.

neshkeev commented:
@smiklosovic , did you decide to do it differently?

neshkeev commented:
@smiklosovic , please let me know why the PR got closed. I don't understand why it happened and how one can be helpful to the project.

smiklosovic commented:
Hi @neshkeev , it was closed because we have committed the work into https://github.com/apache/cassandra/commit/f27790c96912ac9a83f052d8e6d0bfcdfe60ca0e

Once the work has been done, we tend to close the accompanying PR as it just cease to have the reason of its existence.
